crates/brioche-core/Cargo.toml

```diff
@@ -71,12 +71,15 @@ object_store = { git = "https://github.com/apache/arrow-rs-object-store.git", re
 ] }
 opentelemetry = "0.30.0"
 opentelemetry-otlp = { version = "0.30.0", default-features = false, features = [
+    "logs",
+    "trace",
     "http-proto",
     "reqwest-blocking-client",
     "reqwest-rustls",
 ] }
+opentelemetry-appender-tracing = { version = "0.30.0", features = ["experimental_use_tracing_span_context"] }
```

Comment:
The main logic of this PR comes from this crate which does the bridge between `tracing` and `opentelemetry-otlp`. The feature `experimental_use_tracing_span_context` is here to associate logs with traces by associating a trace ID to each log.